### ISSUE TITLE
Tensor.t() should throw an exception on non-2D tensors

### ISSUE THREAD

**Issue by colesbury** [pytorch/pytorch#7010]
PyTorch 0.4 and master

```python
import torch
x = torch.randn(3, 5, 6)
print(x.t().shape)
print(x.t().stride())
```

Previously, Tensor.t() threw an exception on non-2D tensors.

**Comment by zou3519** [comment]
It looks like all of the `before_call` options in Declarations.cwrap stopped working (I tried grepping the generated code for some of them but could not find any).
![image](https://user-images.githubusercontent.com/5652049/39369925-8974c944-4a0b-11e8-8a44-447e07d905b8.png)

**Comment by li-roy** [comment]
This bug was introduced in the variable-tensor merge. It seems that all cwrap plugins broke on that commit.

**Comment by gchanan** [comment]
I'm going to take a look at this.

**Comment by gchanan** [comment]
Some of these don't make sense, like with_gil / auto_gpu can't work now because ATen doesn't know about gils / auto_gpu; we handle those in the binding code.  So we should get rid of the ones that don't make sense and fix the ones that do :).

**Comment by zou3519** [comment]
@gchanan sync with @li-roy, he took a look at it yesterday and might have insights

**Comment by li-roy** [comment]
Closed by #7200
